Automattic/newspack-plugin
# Fix TypeScript use; add to CI

### All Submissions:






### Changes proposed in this Pull Request:



Some TS was added in this project, but a CI step to keep it checked was not added. Since we don't _compile_ TS, but only _verify_ it's validity, the TS code was invalid but we would not know (until verifying locally, e.g. via IDE). This PR adds a CI step to verify TS, fixes* the existing files, and converts a couple of files to TS.

### How to test the changes in this Pull Request:

1. Observe the `check-typescript` CI job passing 👇 
2. Review code changes – there should be no functional changes, just pure refactor and added checks to satisfy TS

### Other information:
